### PR TITLE
feature: edit and detail column in entities table

### PR DESCRIPTION
**Pull request by gregorpirgie**
- includes refactor of related code in views.py, tables.py, and templates
- includes added docstrings

Tables.py
- refactor of tables.py file to own python module within apis_entities
- refactor of get_entities_tables factory method into factory-class.
- factory-class persists already created table-classes, instead of re-creating them on each request.
- inner-generic-table-class removes constraints of former implementation
- its the broadest implementation of the table
- the calling view is now responsible for customizing the instance of the table class that will be send to the template, instead of hardcoding the customization in the class itself.
- added utility class get_entities_table, that uses the new FactoryClass, but catches all
existing calls from other modules, to not break current implementations.

GenericListViewNew in apis_entities.views.py:
- table specific code moved into own class / mixin that is now inherited by the GenericListViewNew
- refactored variable names
- moved seperate logic into own methods
- added docstrings: with context information that is not visible from the code itself.
- tried to name methods in a way that they explain what they do
- added inline comments on class vars and instance vars that are bound to implicit logic that happens in the parent classes - these should NOT be renamed, or things will break.
- removed as many if else statement block-series or try-except blocks that where even interdependent on one another as possible. The prior version was, IMO, really hard to read and reason about.

### CODE REVIEW

**Comment by gregorpirgie** [comment]
@b1rger ignore the empty apis-tei folder, slipped in when I squashed the commits. 

I can remove the folder and, if this PR is approved, squash merge the PR into dev. Same goes for all other small changes we might want to commit during the review.

**Review by b1rger — CHANGES_REQUESTED** [review]
Please, please, stop creating new pull requests! There is really no need to create a new pull request every time you change something in a branch!

> I can remove the folder and, if this PR is approved, squash merge the PR into dev. Same goes for all other small changes we might want to commit during the review.

Why not simply `--amend` the changes to the existing commit?

(Not sure what all those `__gp__` things are - is that a visualstudio thing?)

**Comment by koeaw** [comment]
> Please, please, stop creating new pull requests! There is really no need to create a new pull request every time you change something in a branch!

I strongly agree.

@gregorpirgie all the short-lived duplicate PRs:
- unnecessarily spam the inboxes of everyone subscribed to a repo,
- break the PR review workflow,
- artificially increment our issue count.

You also don't mark any of the old PRs duplicates of their successors, AFAIK (`duplicate` label + backlink), so it's impossible to painlessly trace how individual PRs build on another. 

I can offer an emergency Git rebase crash course next week. Please finally join the GitLab group/repo which I've asked you to join/sent invites for multiples times already. I'll post a handful of possible dates/times on MM.

We can also quickly go over what makes a "good" PR.

@b1rger 
> (Not sure what all those `__gp__` things are - is that a visualstudio thing?)

I'm assuming they are initials, meant to mark a comment as "left by xy". Dunno if the underscores carry any meaning in VSC. A couple of times, I've prefixed TODO comments that I've left with my initials because they are (strong) personal suggestions rather than "hard" TODOs; but yeah, I think they are better kept to a minimum.

**Comment by gregorpirgie** [comment]
> Please, please, stop creating new pull requests! There is really no need to create a new pull request every time you change something in a branch!

sure, will do.

File: apis_core/apis_entities/tables/factory.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
+import warnings
```
[the remark this answers is not in the document]
Comment (gregorpirgie) [reply apis_core/apis_entities/tables/factory.py:1]: I would suggest to keep the warning for now, and implement a general, reusable logger for APIS  as the result of a separate issue. I saw that you already took a step in this direction.

[the remark this answers is not in the document]
Comment (b1rger) [reply apis_core/apis_entities/tables/factory.py:1]: > I would suggest to keep the warning for now, and implement a general, reusable logger for APIS as the result of a separate issue.

There **is** a general, reusable logger implemented in Django. I don't see the benefit of reimplementing something that exists in Django. And even if we decide to do that at some point, until then it make sense to use what is shipped with Django, what the Django documentation suggests.

> I saw that you already took a step in this direction.

I don't know what you are referring to

File: apis_core/apis_entities/tables/factory.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                    return value
+
+            export_formats = [
+                "csv",
+                "json",
+                "xls",
+                "xlsx",
+            ]
+
+            # merge = MergeColumn(verbose_name="keep | remove", accessor="pk")
```
Comment (b1rger) [inline apis_core/apis_entities/tables/factory.py:114]:
```suggestion
```

**Comment by gregorpirgie** [comment]
Will update the PR, and address all the suggestions birger made, so further reviews are not necessary atm - just so that you don't do obsolete work now @koeaw .

File: apis_core/apis_entities/tables/factory.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gregorpirgie) [reply apis_core/apis_entities/tables/factory.py:1]: Clarified this in the conversation already, will use the django logger - if the `__init__` block, where the warning is raised, is still necessary after the next update.

**Comment by b1rger** [comment]
I'm closing this - i think most of issues that the PR tried to address are fixed by now. Thanks anyway!

**Pull request closed without merging**